sklearn/utils/tests/test_plotting.py

```diff
@@ -1,12 +1,122 @@
 import numpy as np
 import pytest
 
+from sklearn.linear_model import LogisticRegression
 from sklearn.utils._plotting import (
+    _BinaryClassifierCurveDisplayMixin,
     _despine,
     _interval_max_min_ratio,
     _validate_score_name,
     _validate_style_kwargs,
 )
+from sklearn.utils._response import _get_response_values_binary
+from sklearn.utils._testing import assert_array_equal
+
+
+@pytest.mark.parametrize("ax", [None, "Ax"])
+@pytest.mark.parametrize("name", [None, "CustomName"])
+def test_validate_plot_params(pyplot, ax, name):
+    """Check `_validate_plot_params` returns the correct values."""
+    import matplotlib.pyplot as plt
+
+    display = _BinaryClassifierCurveDisplayMixin()
+    display.estimator_name = "TestEstimator"
+    if ax:
+        _, ax = plt.subplots()
+    ax_out, _, name_out = display._validate_plot_params(ax=ax, name=name)
+
+    if name:
+        assert name_out == name
+    else:
+        assert name_out == display.estimator_name
+    if ax:
+        assert ax == ax_out
+
+
+@pytest.mark.parametrize("pos_label", [None, 0])
+@pytest.mark.parametrize("name", [None, "CustomName"])
+@pytest.mark.parametrize(
+    "response_method", ["auto", "predict_proba", "decision_function"]
+)
+def test_validate_and_get_response_values(pyplot, pos_label, name, response_method):
+    """Check `_validate_and_get_response_values` returns the correct values."""
+    X = np.array([[0, 0], [1, 1], [2, 2], [3, 3]])
+    y = np.array([0, 0, 2, 2])
+    estimator = LogisticRegression().fit(X, y)
+
+    y_pred, pos_label, name_out = (
+        _BinaryClassifierCurveDisplayMixin._validate_and_get_response_values(
+            estimator,
+            X,
+            y,
+            response_method=response_method,
+            pos_label=pos_label,
+            name=name,
+        )
+    )
+
+    expected_y_pred, expected_pos_label = _get_response_values_binary(
+        estimator, X, response_method=response_method, pos_label=pos_label
+    )
+
+    assert_array_equal(y_pred, expected_y_pred)
```

Comment:
Since we have some float, I would prefer `assert_allclose`.

Reply:
Thanks! Done!